kotest-assertions/kotest-assertions-core/src/commonMain/kotlin/io/kotest/matchers/collections/containExactly.kt

```diff
@@ -45,10 +65,23 @@ fun <T, C : Collection<T>> containExactly(expected: C): Matcher<C?> = neverNullM
       sb.toString()
    }
 
-   MatcherResult(
-      passed,
-      failureMessage
-   ) { "Collection should not be exactly ${expected.printed().value}" }
+   val negatedFailureMessage = { "Collection should not contain exactly ${expected.printed().value}" }
+
+   if (actual.size <= 20 && expected.size <= 20) {
```

Comment:
Maybe this 20 could be extracted to a system property, just for the sake of configurability.